### PR TITLE
Adding PadTo transform

### PR DESCRIPTION
**Pull request by calebrob6**
In the ChesapeakeCVPR trainer I'd like to use a GridGeoSampler in the test dataset (so that I can test over all pixels). However, the patches that are sampled by the GridGeoSampler are all different sizes, therefore I need to pad everything (vs. center crop in training) to some fixed larger size so that they can work with a dataloader.

### CODE REVIEW

**Comment by isaaccorley** [comment]
Nice, I think the only suggestion I would make would be to not force it be constant padding and allow a padding_mode arg. Some recent research I've seen has received better results by using reflective padding.

**Comment by calebrob6** [comment]
Good point, in that case I may allow for separate args for the image vs. the mask padding. Out of curiosity, do you remember the paper that shows that?

**Comment by adamjstewart** [comment]
> However, the patches that are sampled by the GridGeoSampler are all different sizes

Why are they different sizes?

We should discuss this PR in more detail tomorrow. I don't really want to add non-geospatial datasets/transforms/models to torchgeo. These kind of transforms belong in Kornia/torchvision or another library. I haven't yet gotten a chance to play with Kornia and see how to integrate it properly.

**Comment by calebrob6** [comment]
> Why are they different sizes?

They are sampled to be uniform sizes in EPSG:3857 which means different sizes in the local tile coordinate system.

Can we add this until you get the chance to try and train models use transforms? -- it needs to exist somewhere to finish the ChesapeakeCVPR testing (I can just drop in a custom transform function that does this but wanted to try a cleaner route).

**Comment by isaaccorley** [comment]
> Good point, in that case I may allow for separate args for the image vs. the mask padding. Out of curiosity, do you remember the paper that shows that?

Yes, let me try to dig them up out of the abyss that is my Mendeley library.

**Comment by adamjstewart** [comment]
https://github.com/kornia/kornia/issues/1283 and https://github.com/kornia/kornia/pull/1286 are picking up steam so I think it's safe to close this.
